### PR TITLE
Modified coeffs-generator to remove runtime error in solve/nonlinsolve

### LINKED ISSUE #13849: solve/nonlinsolve: RuntimeError: run out of coefficient configurations
```
>>> solve((t*(sqrt(5) + sqrt(2)) - sqrt(2), t), t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\sympy\solvers\solvers.py", line 1157, in solve
    solution = _solve_system(f, symbols, **flags)
  File "sympy\sympy\solvers\solvers.py", line 1766, in _solve_sy
stem
    poly = g.as_poly(*symbols, extension=True)
  File "sympy\sympy\core\basic.py", line 707, in as_poly
    poly = Poly(self, *gens, **args)
  File "sympy\sympy\polys\polytools.py", line 128, in __new__
    return cls._from_expr(rep, opt)
  File "sympy\sympy\polys\polytools.py", line 239, in _from_expr

    return cls._from_dict(rep, opt)
  File "sympy\sympy\polys\polytools.py", line 183, in _from_dict

    domain, rep = construct_domain(rep, opt=opt)
  File "sympy\sympy\polys\constructor.py", line 237, in construc
t_domain
    result = _construct_simple(coeffs, opt)
  File "sympy\sympy\polys\constructor.py", line 44, in _construc
t_simple
    domain, result = _construct_algebraic(coeffs, opt)
  File "sympy\sympy\polys\constructor.py", line 92, in _construc
t_algebraic
    g, span, H = primitive_element(exts, ex=True, polys=True)
  File "sympy\sympy\polys\numberfields.py", line 879, in primiti
ve_element
    raise RuntimeError("run out of coefficient configurations")
RuntimeError: run out of coefficient configurations

>>> nonlinsolve((t*(sqrt(5) + sqrt(2)) - sqrt(2), t), t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\sympy\solvers\solveset.py", line 2188, in nonlinso
lve
    system, symbols)
  File "sympy\sympy\solvers\solveset.py", line 1992, in _separat
e_poly_nonpoly
    poly = eq.as_poly(*symbols, extension=True)
  File "sympy\sympy\core\basic.py", line 707, in as_poly
    poly = Poly(self, *gens, **args)
  File "sympy\sympy\polys\polytools.py", line 128, in __new__
    return cls._from_expr(rep, opt)
  File "sympy\sympy\polys\polytools.py", line 239, in _from_expr

    return cls._from_dict(rep, opt)
  File "sympy\sympy\polys\polytools.py", line 183, in _from_dict

    domain, rep = construct_domain(rep, opt=opt)
  File "sympy\sympy\polys\constructor.py", line 237, in construc
t_domain
    result = _construct_simple(coeffs, opt)
  File "sympy\sympy\polys\constructor.py", line 44, in _construc
t_simple
    domain, result = _construct_algebraic(coeffs, opt)
  File "sympy\sympy\polys\constructor.py", line 92, in _construc
t_algebraic
    g, span, H = primitive_element(exts, ex=True, polys=True)
  File "sympy\sympy\polys\numberfields.py", line 879, in primiti
ve_element
    raise RuntimeError("run out of coefficient configurations")
RuntimeError: run out of coefficient configurations
```

### PR DESCRIPTION
**Pull request by arighnac**
Fixes #13849 .

** NOW **

`>>> from sympy import *`
`>>> t = Symbol('t')`
`>>> solve((t*(sqrt(5) + sqrt(2)) - sqrt(2), t), t)`
`[]`
`>>> nonlinsolve((t*(sqrt(5) + sqrt(2)) - sqrt(2), t), t)`
`EmptySet()`

The above don't show the runtime error as it used to show before, sine the coefficient generator has been modified.

### CODE REVIEW

**Comment by ArighnaIITG** [comment]
@jksuom, @smichr, Please review this PR.

**Comment by jksuom** [comment]
Please add a `test_issue_13849`.

File: sympy/solvers/tests/test_solveset.py [hunk 1]
```diff
@@ -1636,3 +1636,8 @@ def test__is_finite_with_finite_vars():
 
 def test_issue_13550():
     assert solveset(x**2 - 2*x - 15, symbol = x, domain = Interval(-oo, 0)) == FiniteSet(-3)
+
+
+def test_issue_13842():
```
Comment (jksuom) [inline sympy/solvers/tests/test_solveset.py]: This issue does not seem to involve `nonlinsolve`. Is the issue number correct?

Comment (ArighnaIITG) [reply sympy/solvers/tests/test_solveset.py]: Yes, I have changed the typing error. @jksuom

**Comment by jksuom** [comment]
Thanks! I think this will suffice for now.

**Pull request merged**